### ISSUE TITLE
Some states incorrectly return None instead of an empty dict when there are no changes

### ISSUE THREAD

**Issue by OrlandoArcapix** [saltstack/salt#47059]
### Description of Issue/Question

Version 2018.3.0 fails state runs when no changes are made with modules which return "None" instead of an empty dict for the 'changes' data.

See ticket #46985 

### Steps to Reproduce Issue

See the following lines in files in `salt/states/`:
```
boto_cfn.py:        ret['changes'] = None
boto_cfn.py:        ret['changes'] = None
boto_cfn.py:        ret['changes'] = None
boto_iam.py:    ret['changes'] = None
boto_iot.py:            ret['changes'] = None
glassfish.py:            ret['changes'] = None
glassfish.py:            ret['changes'] = None
glassfish.py:        ret['changes'] = None
grafana_datasource.py:            ret['changes'] = None
influxdb_user.py:                ret['changes'] = None
vault.py:        ret['changes'] = None
zenoss.py:        ret['changes'] = None
zenoss.py:        ret['changes'] = None
```

**Comment by OrlandoArcapix** [comment]
Hey @rallytime - I've put in a PR against issue #47059, which replaces those `None` entries with an empty dict`{}`m though I've not been able to test as I don't have an environment which uses the state modules I've updated in that pull request.

**Comment by rallytime** [comment]
Awesome! Thanks @OrlandoArcapix! Let's see how the test run goes on that. Thanks for filing this issue and for the fix. :)

**Comment by rallytime** [comment]
Closed via #47060
